core/store/src/trie/shard_tries.rs

```diff
@@ -81,12 +91,12 @@ impl ShardTries {
     }
 
     pub fn get_store(&self) -> Arc<Store> {
```

Comment:
I'd chage this to 

```rust
pub(crate) fn store(&self) -> &Store {
```

It seems wrong that we have access to the whole `Store` from just `ShardTries`. It seems all call-sites outside of this crate can get access to store from elsewhere.

Reply:
But maybe that's best left to a follow up PR!

Reply:
Agreed